setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsfileRunnerLauncher.java

```diff
@@ -0,0 +1,137 @@
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Set up of Jenkins environment for executing a single Jenkinsfile.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class JenkinsfileRunnerLauncher extends JenkinsEmbedder {
+    private final File warDir;
+    private final File pluginsDir;
+    /**
+     * Keep the reference around to prevent them from getting GCed.
+     */
+    private final Set<Object> noGc = new HashSet<>();
+
+    public JenkinsfileRunnerLauncher(File warDir, File pluginsDir) {
+        this.warDir = warDir;
+        this.pluginsDir = pluginsDir;
+    }
+
+    /**
+     * Sets up Jetty without any actual TCP port serving HTTP.
+     */
+    @Override
+    protected ServletContext createWebServer() throws Exception {
+        QueuedThreadPool queuedThreadPool = new QueuedThreadPool(10);
+        server = new Server(queuedThreadPool);
+
+        WebAppContext context = new WebAppContext(warDir.getPath(), contextPath);
+        context.setClassLoader(getClass().getClassLoader());
+        context.setConfigurations(new Configuration[]{new WebXmlConfiguration()});
+        context.addBean(new NoListenerConfiguration(context));
+        server.setHandler(context);
+        context.getSecurityHandler().setLoginService(configureUserRealm());
+        context.setResourceBase(warDir.getPath());
+
+        server.start();
+
+        localPort = -1;
+
+        setPluginManager(new PluginManagerImpl(context.getServletContext(),pluginsDir));
+
+        return context.getServletContext();
+    }
+
+    @Override
+    public void recipe() throws Exception {
+        // TODO
+    }
+
+    /**
+     * Supply a dummy {@link LoginService} that allows nobody.
+     */
+    @Override
+    protected LoginService configureUserRealm() {
+        return new HashLoginService();
+    }
+
+    @Override
+    public void before() throws Throwable {
+        setLogLevels();
+        super.before();
+    }
+
+    /**
+     * We don't want to clutter console with log messages, so kill of any unimportant ones.
+     */
+    private void setLogLevels() {
+        Logger.getLogger("").setLevel(Level.WARNING);
+        // Prevent warnings for plugins with old plugin POM (JENKINS-54425)
+        Logger.getLogger(ClassicPluginStrategy.class.getName()).setLevel(Level.SEVERE);
+        Logger l = Logger.getLogger(DeprecatedAgentProtocolMonitor.class.getName());
+        l.setLevel(Level.OFF);
+        noGc.add(l);    // the configuration will be lost if Logger gets GCed.
+    }
+
+    /**
+     * Skips the clean up.
+     *
+     * This was initially motivated by SLF4J leaving gnarly messages.
+     * The whole JVM is going to die anyway, so we don't really care about cleaning up anything nicely.
+     */
+    @Override
+    public void after() throws Exception {
+        jenkins = null;
+        super.after();
+    }
+
+    //TODO: add support of timeout
```

Comment:
Created https://issues.jenkins-ci.org/browse/JENKINS-55295

```suggestion
    //TODO: add support of timeout (JENKINS-55295)
```